benchmark/serializer.py

```diff
@@ -48,14 +49,22 @@ def serialize(dataset_name, dataset_path, serialized_graph_path):
 
     for s in serialize_queries:
         logging.info('Executing query: %s', s)
-        try:
-            # Run kuzu shell one query at a time. This ensures a new process is
-            # created for each query to avoid memory leaks.
-            subprocess.run([kuzu_exec_path, serialized_graph_path],
-                           input=(s + ";" + "\n").encode("ascii"), check=True)
-        except subprocess.CalledProcessError as e:
+        # Run kuzu shell one query at a time. This ensures a new process is
+        # created for each query to avoid memory leaks.
+        process = subprocess.Popen([kuzu_exec_path, serialized_graph_path],
+            stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+        process.stdin.write((s + ";" + "\n").encode("ascii"))
+        process.stdin.close()
+        match = re.match('copy (.*) from', s, re.IGNORECASE)
+        for line in iter(process.stdout.readline, b''):
+            print(line.decode("utf-8"), end='')
+            if match:
+                with open(os.path.join(benchmark_copy_log_dir, match.group(1) + '_log.txt'), 'ab') as f:
```

Comment:
You should consider move this out of the `for` loop. You should not need to reopen the file each time.